File: ros_homework/src/something_about_topic/scripts/mag_subscriber.py
#!/usr/bin/env python3

import rospy
from something_about_topic.msg import mag

def mag_callback(msg):
    rospy.loginfo("Received mag data[magnetic_field: (%f, %f, %f)]", 
                  msg.magnetic_field.x, msg.magnetic_field.y, msg.magnetic_field.z)

def mag_subscriber():
    rospy.init_node('mag_subscriber', anonymous=True)
    rospy.Subscriber('/driver/mag', mag, mag_callback)
    rospy.spin()

if __name__ == '__main__':
    try:
        mag_subscriber()
    except rospy.ROSInterruptException:
        pass

File: ros_homework/src/something_about_topic/src/mag_subscriber.cpp
#include <ros/ros.h>
#include <something_about_topic/mag.h>

void magCallback(const something_about_topic::mag::ConstPtr& msg) {
  ROS_INFO("Received mag data[magnetic_field: (%f, %f, %f)]", 
           msg->magnetic_field.x, msg->magnetic_field.y, msg->magnetic_field.z);
}


int main(int argc, char **argv) {
  ros::init(argc, argv, "mag_subscriber");
  ros::NodeHandle n;
  ros::Subscriber mag_sub = n.subscribe("/driver/mag", 10, magCallback);

  ros::spin();

  return 0;
}